presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java

```diff
@@ -0,0 +1,206 @@
+            QueryContext queryContext = task.getQueryContext();
+            if (!queriesToActiveTasks.containsKey(queryContext.getQueryId())) {
+                queriesToActiveTasks.put(queryContext.getQueryId(), new ArrayList<>());
+            }
+
+            queriesToActiveTasks.get(queryContext.getQueryId()).add(task);
+        }
+
+        return queriesToActiveTasks;
+    }
+
+    private List<SqlTask> getActiveSqlTasks()
+    {
+        ArrayList<SqlTask> sqlTasks = new ArrayList<>();
+        for (SqlTask task : sqlTaskManager.getAllTasks()) {
+            if (!task.getTaskInfo().getTaskStatus().getState().isDone()) {
+                sqlTasks.add(task);
+            }
+        }
+
+        return sqlTasks;
+    }
+
+    private void logInfoTable(List<List<String>> table)
+    {
+        for (String row : StringTableUtils.getTableStrings(table)) {
+            log.info(row);
+        }
+    }
+
+    private void logQueryInfo(QueryContext ctx)
+    {
+        logInfoTable(Arrays.asList(
+                Arrays.asList(
+                        "Query ID",
+                        "Max User Bytes",
+                        "Max Total Bytes"),
+                Arrays.asList(
+                        ctx.getQueryId().toString(),
+                        Long.toString(ctx.getMaxUserMemory()),
+                        Long.toString(ctx.getMaxTotalMemory()))));
+    }
+
+    private void logTaskInfos(List<SqlTask> sqlTasks)
+    {
+        ArrayList<List<String>> taskLogInfoList = new ArrayList<>();
+        taskLogInfoList.add(Arrays.asList(
+                "Task ID",
+                "State",
+                "Created Ts",
+                "User Memory",
+                "System Memory",
+                "Input Bytes",
+                "Output Bytes",
+                "Input Row Count",
+                "Output Row Count"));
+
+        for (SqlTask task : sqlTasks) {
+            TaskInfo taskInfo = task.getTaskInfo();
+            SqlTaskIoStats taskIOStats = task.getIoStats();
+            TaskStatus taskStatus = taskInfo.getTaskStatus();
+            TaskStats taskStats = taskInfo.getStats();
+            taskLogInfoList.add(Arrays.asList(
+                    task.getTaskId().toString(),
+                    taskStatus.getState().toString(),
+                    taskStats.getCreateTime().toString(),
+                    taskStats.getUserMemoryReservation().toString(),
+                    taskStats.getSystemMemoryReservation().toString(),
+                    Long.toString(taskIOStats.getInputDataSize().getTotalCount()),
+                    Long.toString(taskIOStats.getOutputDataSize().getTotalCount()),
+                    Long.toString(taskIOStats.getInputPositions().getTotalCount()),
+                    Long.toString(taskIOStats.getOutputPositions().getTotalCount())));
+        }
+
+        logInfoTable(taskLogInfoList);
+    }
+
+    private void logTaskInfosByQueryID(Map<QueryId, List<SqlTask>> queryIDToSqlTaskMap)
```

Comment:
In the Presto projects (as in many Java projects) we use reverse method ordering. If method `A` is called by method `B`, the method ordering will be `B` -> `A`. Please reorder.